### PR TITLE
In-article notifications.

### PR DESCRIPTION
**Pull request by dbrant**
https://phabricator.wikimedia.org/T287113
https://phabricator.wikimedia.org/T285765
https://phabricator.wikimedia.org/T288248

### CODE REVIEW

File: app/src/main/java/org/wikipedia/analytics/NotificationsABCTestFunnel.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    fun logShow() {
+        val item = when (aBTestGroup) {
+            0 -> "inbox"
+            1 -> "bell"
+            else -> "more"
+        }
+        log(
+            "action", "show",
+            "menuItem", item
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/analytics/NotificationsABCTestFunnel.kt:19]: Would it be possible to have these strings as `const` variables?

File: app/src/main/java/org/wikipedia/page/PageActivity.kt [hunk 2]
```diff
@@ -690,9 +701,65 @@ class PageActivity : BaseActivity(), PageFragment.Callback, LinkPreviewDialog.Ca
...
+    private fun setupNotificationsTest() {
+        binding.pageToolbarButtonNotifications.isVisible = false
+        binding.unreadDotView.isVisible = false
+        when (notificationsABCTestFunnel.aBTestGroup) {
+            0 -> {
+                binding.pageToolbarButtonNotifications.setIcon(R.drawable.ic_inbox_24)
+            }
+            1 -> {
+                binding.pageToolbarButtonNotifications.setIcon(R.drawable.ic_notifications_black_24dp)
+            }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/page/PageActivity.kt:718]: Minor but the parentheses can be removed.

File: app/src/main/res/xml/developer_preferences.xml [hunk 3]
```diff
@@ -349,6 +349,11 @@
 
     <PreferenceCategory android:title="A/B tests">
 
+        <org.wikipedia.settings.IntPreference
+            style="@style/IntPreference"
+            android:key="ab_test_notificationIcons"
+            android:title="Test group for in-app Notification icons (0, 1, or 2)" />
```
Comment (cooltey) [inline app/src/main/res/xml/developer_preferences.xml:355]: I am seeing my test group id is `-1519020059`, which is a negative number. Is this expected?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/analytics/NotificationsABCTestFunnel.kt (hunk 1 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/analytics/NotificationsABCTestFunnel.kt]: sure!

File: app/src/main/java/org/wikipedia/page/PageActivity.kt (hunk 2 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/page/PageActivity.kt]: ok!

File: app/src/main/res/xml/developer_preferences.xml (hunk 3 above)
Comment (dbrant) [reply app/src/main/res/xml/developer_preferences.xml:355]: Yes, that is fine. The test group is your group id _modulo 3_.

**Pull request merged**